### PR TITLE
disable (most) warnings in generated files

### PR DESCRIPTION
**Pull request by murcake**
Some user-enabled warnings (missing-export-lists,
     prepositive-qualified) don't make sense to check in generated
     files, but it takes effort to conform to them.

The solution is to disable all such warnings.

### CODE REVIEW

**Review by sol — APPROVED** [review]
This looks good.

To get this released to Hackage this are the things I want to see:

- [ ] An entry to the change log
- [ ] A version bump; make sure that you have the latest version of `hpack` on your PATH then run [`util/bump-version`](https://github.com/hspec/hspec/blob/main/util/bump-version)
- [ ] Please also revert https://github.com/hspec/hspec/commit/ffd91356901b2a60bd1f90112b61603348b3df36. I think it’s redundant now.
- [ ] I think we only need one of `test-suite with-warn-prepositive-qualified-module` or `test-suite with-warn-missing-export-lists`; keep the one you prefer, remove the other
- [ ] Add ` fixes #702` (or any of the alternatives that GitHub understands) to your commit message so that the issue will be closed automatically

File: hspec-discover/integration-test/hspec-discover-integration-tests.cabal [hunk 1]
```diff
@@ -142,6 +142,23 @@ test-suite with-warn-missing-export-lists
...
   default-language: Haskell2010
 
+test-suite with-warn-prepositive-qualified-module
+  build-tool-depends: hspec-discover:hspec-discover
+  type:
+      exitcode-stdio-1.0
+  ghc-options:
+      -Wprepositive-qualified-module -Werror
+  if !impl(ghc >= 8.10)
+    buildable: false
```
Comment (sol) [inline hspec-discover/integration-test/hspec-discover-integration-tests.cabal:152]: Apparently somebody decided that booleans in `.cabal` files should be case sensitive now. 

https://github.com/hspec/hspec/runs/8173846384?check_suite_focus=true#step:4:310

So I guess for for compatibility it makes sense to use `False` here.

**Comment by murcake** [comment]
> This looks good.
> 
> To get this released to Hackage this are the things I want to see:
> 
>     * [ ]  An entry to the change log
> 
>     * [ ]  A version bump; make sure that you have the latest version of `hpack` on your PATH then run [`util/bump-version`](https://github.com/hspec/hspec/blob/main/util/bump-version)
> 
>     * [ ]  Please also revert [ffd9135](https://github.com/hspec/hspec/commit/ffd91356901b2a60bd1f90112b61603348b3df36). I think it’s redundant now.
> 
>     * [ ]  I think we only need one of `test-suite with-warn-prepositive-qualified-module` or `test-suite with-warn-missing-export-lists`; keep the one you prefer, remove the other
> 
>     * [ ]  Add ` fixes #702` (or any of the alternatives that GitHub understands) to your commit message so that the issue will be closed automatically

I'm trying to build hpack-0.35 and get ghc panic

> This looks good.
> 
> To get this released to Hackage this are the things I want to see:
> 
>     * [ ]  An entry to the change log
> 
>     * [ ]  A version bump; make sure that you have the latest version of `hpack` on your PATH then run [`util/bump-version`](https://github.com/hspec/hspec/blob/main/util/bump-version)
> 
>     * [ ]  Please also revert [ffd9135](https://github.com/hspec/hspec/commit/ffd91356901b2a60bd1f90112b61603348b3df36). I think it’s redundant now.
> 
>     * [ ]  I think we only need one of `test-suite with-warn-prepositive-qualified-module` or `test-suite with-warn-missing-export-lists`; keep the one you prefer, remove the other
> 
>     * [ ]  Add ` fixes #702` (or any of the alternatives that GitHub understands) to your commit message so that the issue will be closed automatically

I believe I've done all of that

**Review by sol — APPROVED** [review]
This looks good to me 👍

Not sure what’s going on with the Windows build.

**Comment by murcake** [comment]
@sol can you please approve, I'm proving to see how CI can be fixed?

**Comment by murcake** [comment]
I'll remove the WIP commits, once it works

**Comment by sol** [comment]
@murcake passing now, apparently related to filename lengths.

**Comment by murcake** [comment]
> @murcake passing now, apparently related to filename lengths.

Hilarious every time :D

Then I'll leave the current short name and fixup. Any other suggestions, or good to merge after that?

**Comment by sol** [comment]
> > @murcake passing now, apparently related to filename lengths.
> 
> Hilarious every time :D

Indeed!

> Then I'll leave the current short name and fixup. Any other suggestions, or good to merge after that?

Just squash the wip commit, which you already did. So good to go.

Thanks for working on this.

**Pull request merged**